### PR TITLE
feat: add support for `approval_requested` reason

### PR DESCRIPTION
**Pull request by setchy**
As per https://docs.github.com/en/rest/activity/notifications?apiVersion=2022-11-28#about-notification-reasons

### CODE REVIEW

**Comment by afonsojramos** [comment]
If we are adding all of these new notifications, should we have a more granular control of what notifications gitify fetches to the user?

File: src/utils/github-api.ts [hunk 1]
```diff
@@ -24,21 +24,22 @@ import { CheckSuiteStatus, Reason, Subject } from '../typesGithub';
 
 // prettier-ignore
```
Comment (afonsojramos) [inline src/utils/github-api.ts:25]: Why are we ignoring prettier here?

File: src/utils/github-api.ts [hunk 2]
```diff
@@ -47,6 +48,8 @@ export function formatReason(reason: Reason): {
 } {
   // prettier-ignore
```
Comment (afonsojramos) [inline src/utils/github-api.ts:49]: Same here

File: src/utils/github-api.ts (hunk 1 above)
Comment (setchy) [reply src/utils/github-api.ts:25]: 🤷 pre-existing. Can do a sweep in a follow up PR

Comment (setchy) [reply src/utils/github-api.ts:25]: pushed a new commit which only adds the new description... not sure how the others got reformatted

**Comment by setchy** [comment]
> If we are adding all of these new notifications, should we have a more granular control of what notifications gitify fetches to the user?
> Just pointing out that we should probably roll that feature out before we roll out a new version.

Perhaps... there is a feature request ticket already for filtering by reason...

Note, we aren't adding ANY new notifications actually, what I've been doing is ensuring that we have fewer Unknown notifications that cannot be clicked on.

**Review by afonsojramos — APPROVED** [review]

**Pull request merged**